spec/module_validator_spec.rb

```diff
@@ -0,0 +1,227 @@
+          "References NOCVE please include NOCVE values in the 'notes' section, rather than in 'references'.",
+          "References AKA please include AKA values in the 'notes' section, rather than in 'references'."
+        ]
+      end
+    end
+
+    context 'when the license contains an invalid value' do
+      let(:mod_options) do
+        super().merge(license: 'MSF_LICENSE')
+      end
+
+      it 'has errors' do
+        expect(subject.errors.full_messages).to eq ['License must include a valid license']
+      end
+    end
+
+    context 'when the rank contains an invalid value' do
+      let(:mod_options) do
+        super().merge(rank: 'ExcellentRanking')
+      end
+
+      it 'has errors' do
+        expect(subject.errors.full_messages).to eq ['Rank must include a valid module ranking']
+      end
+    end
+
+    context 'when the author is missing' do
+      let(:mod_options) do
+        super().merge(author: [])
+      end
+
+      it 'has errors' do
+        expect(subject.errors.full_messages).to eq ["Author can't be blank"]
+      end
+    end
+
+    context 'when the author contains bad characters' do
+      let(:mod_options) do
+        super().merge(author: [
+          Msf::Author.new('@Foobar'),
+          Msf::Author.new('Foobar')
+        ])
+      end
+
+      it 'has errors' do
+        expect(subject.errors.full_messages).to eq ['Author must not include username handles, found "@Foobar". Try leaving it in a comment instead.']
+      end
+    end
+
+    context 'when the module name contains bad characters' do
+      let(:mod_options) do
+        super().merge(name: 'Testing <> bad & chars')
+      end
+
+      it 'has errors' do
+        expect(subject.errors.full_messages).to eq ['Name must not contain the characters &<>']
+      end
+    end
+
+    context 'when the module file path is not snake case' do
+      let(:mod_options) do
+        super().merge(file_path: 'modules/exploits/windows/smb/CVE_2020_0796_smbghost.rb')
+      end
+
+      it 'has errors' do
+        expect(subject.errors.full_messages).to eq ['File path must be snake case, instead found "modules/exploits/windows/smb/CVE_2020_0796_smbghost.rb"']
+      end
+    end
+
+    context 'when the description is missing' do
+      let(:mod_options) do
+        super().merge(description: nil)
+      end
+
+      it 'has errors' do
+        expect(subject.errors.full_messages).to eq ["Description can't be blank"]
+      end
+    end
+
+    context 'when the platform value is invalid', skip_before: true do
+      let(:mod_options) do
+        super().merge(platform: Msf::Module::PlatformList.new('foo'))
+      end
+
+      it 'raises an ArgumentError' do
+        expect { subject }.to raise_error ArgumentError, 'No classes in Msf::Module::Platform for foo!'
+      end
+    end
+
```

Comment:
Not a blocker: We _could_ write a separate test here to handle invalid platform list values that doesn't rely on the external behavior of `Msf::Module::PlatformList` raising an exception:


```ruby
      let(:mod_options) do
        super().merge(
            platform: [
              instance_double(Msf::Module::Platform, realname: 'Foo')
            ]
        )
      end
```

Reply:
Will resolve this; Let's hope the invariant holds true in the future.